perf(rtp): reduce rtp scans (#24191)
neovim/neovim: runtime/doc/repeat.txt

				:runtime plugin/**/*.vim
<			would source the first file only.

> 			For each {file} pattern, if a file has both a `.vim`
> 			and `.lua` extensions, the `.vim` version will take
> 			priority and be sourced first.

Reviewer: ```suggestion
			For each {file} pattern, if both `.vim` and `.lua`
			files match in a directory, the `.vim` files take
			priority and are sourced first.
```

Author: This is stronger than what is intended. We just want to enforce that `foo/bar.vim` take priority over `foo/bar.lua`.

Reviewer: At least add the "in a directory"

Reviewer: How about this
```suggestion
			For each {file} pattern, if a multiple matches differ
			only in extension, the `.vim` version is sourced first
			the `.lua` version second, and others last.
```

> 			and `.lua` extensions, the `.vim` version will take
> 			priority and be sourced first.

Reviewer 2: ```suggestion
			and `.lua` extensions, the `.vim` version will be sourced first.
```

"Priority" is whatever is sourced last, right? So maybe it's less confusing to not mention priority here.
